### PR TITLE
HARP-7292: Improve tile cache use.

### PR DESCRIPTION
**Pull request by kkostecki**
Thank you for contributing to harp.gl!

Before requesting a pull request, please remember to check the following documents:
* [contribution guidelines](https://github.com/heremaps/harp.gl/blob/master/CONTRIBUTING.md)
* [coding style](https://github.com/heremaps/harp.gl/blob/master/CODINGSTYLE.md)

If you are adding new functionality we would highly appreciate if you can describe what is the capability you are adding and even better if you can add some examples. Please also remember to add tests for it.

# CI Check

Our bots will check whether your PR can be directly integrated into the mainline. We have some internal integration tests running on the background, our bots will inform you of the next steps and someone from our team will take a look and help if needed!

And please do not forget to sign-off your commit! You can read more about DCO [here](https://julien.ponge.org/blog/developer-certificate-of-origin-versus-contributor-license-agreements/). But, in short, you just need to use `git commit -s` or append `--signoff` when you are committing to the repo.

Happy contributing!

### CODE REVIEW

**Comment by kkostecki** [comment]
@heremaps-bot retest this please

File: @here/harp-mapview-decoder/lib/TileDataSource.ts [hunk 1]
```diff
@@ -167,6 +167,10 @@ export class TileDataSource<TileType extends Tile> extends DataSource {
         this.useGeometryLoader = true;
         this.cacheable = true;
         this.m_tileLoaderCache = new LRUCache<number, TileLoader>(this.getCacheCount());
+        this.m_tileLoaderCache.evictionCallback = (_, tileLoader) => {
+            // Cancel any pending downloads as early as possible.
+            tileLoader.cancel();
+        };
```
Comment (nzjony) [inline @here/harp-mapview-decoder/lib/TileDataSource.ts]: 👍

File: @here/harp-lrucache/lib/LRUCache.ts [hunk 2]
```diff
@@ -66,10 +66,15 @@ export class LRUCache<Key, Value> {
     private m_sizeFunction: (v: Value) => number;
 
     /**
-     * Creates a new instance of `LRUCache`. The optional sizeFunction can be used
-     * fine tune the size required to cache that item.
+     * Creates a new instance of `LRUCache`.
      *
-     * @param cacheCapacity The maximum number of entries to store in the cache.
+     * The optional [[sizeFunction]] can be used to fine tune the memory consuption of all cached
+     * elements, thus [[cacheCapacity]] is then a meassure of memory used. Otherwise, if
```
[the remark this answers is not in the document]
Comment (kkostecki) [reply @here/harp-lrucache/lib/LRUCache.ts]: Done.

File: @here/harp-lrucache/lib/LRUCache.ts [hunk 3]
```diff
@@ -66,10 +66,15 @@ export class LRUCache<Key, Value> {
...
+     * [[sizeFunction]] is not specified, the [[cacheCapacity]] amounts for the number of elements
+     * stored.
+     *
+     * @param cacheCapacity The maximum number of entries to store in the cache if [[sizeFunction]]
```
[the remark this answers is not in the document]
Comment (kkostecki) [reply @here/harp-lrucache/lib/LRUCache.ts]: Done.

File: @here/harp-lrucache/lib/LRUCache.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (kkostecki) [reply @here/harp-lrucache/lib/LRUCache.ts]: Done.

**Comment by kkostecki** [comment]
@heremaps-bot retest this please

File: @here/harp-lrucache/lib/LRUCache.ts [hunk 4]
```diff
@@ -66,10 +66,15 @@ export class LRUCache<Key, Value> {
     private m_sizeFunction: (v: Value) => number;
 
     /**
-     * Creates a new instance of `LRUCache`. The optional sizeFunction can be used
-     * fine tune the size required to cache that item.
+     * Creates a new instance of `LRUCache`.
      *
-     * @param cacheCapacity The maximum number of entries to store in the cache.
+     * The optional [[sizeFunction]] can be used to fine tune the memory consuption of all cached
```
Comment (ninok) [inline @here/harp-lrucache/lib/LRUCache.ts]: typo: consumption

File: @here/harp-lrucache/lib/LRUCache.ts [hunk 5]
```diff
@@ -66,10 +66,15 @@ export class LRUCache<Key, Value> {
...
+     * elements, thus [[cacheCapacity]] means then memory used (in MBs). Otherwise, if
+     * [[sizeFunction]] is not specified, the [[cacheCapacity]] accounts for the maximum
+     * number of elements stored.
+     *
+     * @param cacheCapacity Number used to configure the maximum cache size, may express
+     * number of entries or memory consumed in megabytes depending from [[sizeFunction]].
```
Comment (ninok) [inline @here/harp-lrucache/lib/LRUCache.ts]: Typo: depending on

File: @here/harp-lrucache/lib/LRUCache.ts (hunk 4 above)
Comment (kkostecki) [reply @here/harp-lrucache/lib/LRUCache.ts]: Fixed

File: @here/harp-lrucache/lib/LRUCache.ts (hunk 5 above)
Comment (kkostecki) [reply @here/harp-lrucache/lib/LRUCache.ts]: Fixed

**Pull request merged**